src/llm_vm/completion/data_synthesis.py

```diff
@@ -49,12 +49,28 @@ def data_synthesis(self, optimizer, prompt, response, openai_key=None, completio
         return datapoints
     
     @backoff.on_exception(backoff.expo, openai.error.RateLimitError)
-    def generate_example(self, final_prompt, openai_key, example_delim="<END>", model="gpt-4", max_tokens=1000, temperature=1, completion=None):
+    def generate_example(
+        self, 
+        optimizer, 
+        final_prompt, 
+        openai_key, 
+        example_delim="<END>", 
+        max_length=1000, 
+        temperature=1, # temperature 1 ideal for generating examples (default is 0)
+        completion=None, 
+        **kwargs
+        ):
         openai.api_key = openai_key
-        cur_prompt = [{'role': "system", 'content': final_prompt}]
+        kwargs.update({"temperature":temperature})
+
+        response = optimizer.big_model.generate(
```

Comment:
@cartazio this is the main fix- calling the `big_model` attribute of the optimizer for generation instead of openai api

Reply:
looks good. if you can undo the formatting changes then i can merge this in